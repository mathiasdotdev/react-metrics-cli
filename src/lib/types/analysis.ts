/**
 * Type de déclaration détectée dans le code
 */
export enum DeclarationType {
  CONSTANT = 'CONSTANT',
  FUNCTION = 'FUNCTION',
  CLASS = 'CLASS',
  CONSOLE = 'CONSOLE',
  PROP = 'PROP',
  EXPORT = 'EXPORT',
  DEPENDENCY = 'DEPENDENCY',
  DEFINITION = 'DEFINITION',
}

/**
 * Localisation d'une déclaration dans un fichier
 */
export interface Location {
  filePath: string;
  line: number;
  column: number;
}

/**
 * Représente une déclaration trouvée dans le code
 */
export interface Declaration {
  name: string;
  type: DeclarationType;
  location: Location;
  isUsedLocally: boolean;
  isImportedExternally: boolean;
  isDeprecated?: boolean;
  context?: string;
  metadata?: Record<string, unknown>;
}

/**
 * Résultat complet de l'analyse
 */
export interface AnalysisResult {
  declarations: Map<string, Declaration>;
  deadCode: Declaration[];
  deprecated: Declaration[];
  duration: number;
}

/**
 * Génère une clé unique pour une déclaration
 */
export const getDeclarationKey = (declaration: Declaration): string => {
  return `${declaration.location.filePath}:${declaration.location.line}:${declaration.location.column}:${declaration.name}`;
};

/**
 * Vérifie si une déclaration est utilisée (localement ou importée ailleurs)
 */
export const isUsed = (declaration: Declaration): boolean => {
  return declaration.isUsedLocally || declaration.isImportedExternally;
};

/**
 * Récupère le code mort (déclarations non utilisées)
 */
export const getDeadCode = (
  declarations: Map<string, Declaration>,
): Declaration[] => {
  return Array.from(declarations.values()).filter((d) => !isUsed(d));
};

/**
 * Récupère toutes les déclarations marquées @deprecated
 */
export const getDeprecated = (
  declarations: Map<string, Declaration>,
): Declaration[] => {
  return Array.from(declarations.values())
    .filter((d) => d.isDeprecated === true)
    .sort((a, b) => a.location.filePath.localeCompare(b.location.filePath));
};
