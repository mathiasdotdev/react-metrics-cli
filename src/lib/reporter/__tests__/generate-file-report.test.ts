import { AnalysisResult, Declaration, DeclarationType } from '$types/analysis';
import { describe, expect, it } from 'bun:test';
import {
  buildDetailedReport,
  generateFileReport,
} from '../generate-file-report';
import * as fs from 'fs/promises';
import * as path from 'path';

describe('generateFileReport', () => {
  const createDeclaration = (
    name: string,
    type: DeclarationType,
    filePath: string,
    line: number,
    column: number,
    usedLocally: boolean = false,
    importedExternally: boolean = false,
  ): Declaration => ({
    name,
    type,
    location: { filePath, line, column },
    isUsedLocally: usedLocally,
    isImportedExternally: importedExternally,
  });

  /**
   * Normalise le rapport en remplaçant les dates et durées variables
   */
  const normalizeReport = (report: string): string => {
    return report
      .replace(
        /Généré le:\s+\d{2}\/\d{2}\/\d{4}\s+\d{2}:\d{2}:\d{2}/g,
        'Généré le: DD/MM/YYYY HH:MM:SS',
      )
      .replace(
        /Durée totale de l'analyse:\s+\d+\.\d+\s+secondes?/g,
        "Durée totale de l'analyse: X.XX secondes",
      );
  };

  it('should generate empty report', () => {
    const result: AnalysisResult = {
      declarations: new Map(),
      deadCode: [],
      deprecated: [],
      duration: 500,
    };

    const report = normalizeReport(buildDetailedReport(result));
    expect(report).toMatchSnapshot();
  });

  it('should generate report with functions only', () => {
    const declarations = new Map<string, Declaration>();
    const fn1 = createDeclaration(
      'unusedFunction',
      DeclarationType.FUNCTION,
      '/src/file.ts',
      10,
      5,
    );
    const fn2 = createDeclaration(
      'anotherUnused',
      DeclarationType.FUNCTION,
      '/src/other.ts',
      20,
      3,
    );
    declarations.set('fn1', fn1);
    declarations.set('fn2', fn2);

    const result: AnalysisResult = {
      declarations,
      deadCode: [fn1, fn2],
      duration: 1000,
    };

    const report = normalizeReport(buildDetailedReport(result));
    expect(report).toMatchSnapshot();
  });

  it('should generate report with classes only', () => {
    const declarations = new Map<string, Declaration>();
    const cls1 = createDeclaration(
      'UnusedClass',
      DeclarationType.CLASS,
      '/src/class.ts',
      15,
      7,
    );
    const cls2 = createDeclaration(
      'OldClass',
      DeclarationType.CLASS,
      '/src/legacy.ts',
      30,
      7,
    );
    declarations.set('cls1', cls1);
    declarations.set('cls2', cls2);

    const result: AnalysisResult = {
      declarations,
      deadCode: [cls1, cls2],
      duration: 1000,
    };

    const report = normalizeReport(buildDetailedReport(result));
    expect(report).toMatchSnapshot();
  });

  it('should generate report with constants and props', () => {
    const declarations = new Map<string, Declaration>();
    const c1 = createDeclaration(
      'UNUSED_CONST',
      DeclarationType.CONSTANT,
      '/src/constants.ts',
      5,
      7,
    );
    const p1 = createDeclaration(
      'unusedProp',
      DeclarationType.PROP,
      '/src/props.tsx',
      8,
      3,
    );
    declarations.set('c1', c1);
    declarations.set('p1', p1);

    const result: AnalysisResult = {
      declarations,
      deadCode: [c1, p1],
      duration: 1000,
    };

    const report = normalizeReport(buildDetailedReport(result));
    expect(report).toMatchSnapshot();
  });

  it('should generate report with mixed categories', () => {
    const declarations = new Map<string, Declaration>();
    const fn = createDeclaration('fn', DeclarationType.FUNCTION, '/f.ts', 1, 1);
    const cls = createDeclaration('cls', DeclarationType.CLASS, '/c.ts', 1, 1);
    const constant = createDeclaration(
      'const1',
      DeclarationType.CONSTANT,
      '/c.ts',
      1,
      1,
    );
    const prop = createDeclaration('prop', DeclarationType.PROP, '/p.ts', 1, 1);
    const console = createDeclaration(
      'console',
      DeclarationType.CONSOLE,
      '/c.ts',
      1,
      1,
    );
    const type = createDeclaration(
      'Type1',
      DeclarationType.DEFINITION,
      '/t.ts',
      1,
      1,
    );
    const exp = createDeclaration(
      'Export1',
      DeclarationType.EXPORT,
      '/e.ts',
      1,
      1,
    );
    const dep = createDeclaration(
      'dep',
      DeclarationType.DEPENDENCY,
      '/p.json',
      1,
      1,
    );

    declarations.set('fn', fn);
    declarations.set('cls', cls);
    declarations.set('constant', constant);
    declarations.set('prop', prop);
    declarations.set('console', console);
    declarations.set('type', type);
    declarations.set('exp', exp);
    declarations.set('dep', dep);

    const result: AnalysisResult = {
      declarations,
      deadCode: [fn, cls, constant, prop, console, type, exp, dep],
      duration: 2500,
    };

    const report = normalizeReport(buildDetailedReport(result));
    expect(report).toMatchSnapshot();
  });

  it('should generate report with real-world scenario', () => {
    const declarations = new Map<string, Declaration>();
    const deadCode: Declaration[] = [];

    // 10 Functions
    for (let i = 0; i < 10; i++) {
      const fn = createDeclaration(
        `function${i}`,
        DeclarationType.FUNCTION,
        '/f.ts',
        i + 1,
        1,
      );
      declarations.set(`fn${i}`, fn);
      deadCode.push(fn);
    }

    // 3 Classes
    for (let i = 0; i < 3; i++) {
      const cls = createDeclaration(
        `Class${i}`,
        DeclarationType.CLASS,
        '/c.ts',
        i + 1,
        1,
      );
      declarations.set(`cls${i}`, cls);
      deadCode.push(cls);
    }

    // 12 Constants
    for (let i = 0; i < 12; i++) {
      const c = createDeclaration(
        `const${i}`,
        DeclarationType.CONSTANT,
        '/c.ts',
        i + 1,
        1,
      );
      declarations.set(`const${i}`, c);
      deadCode.push(c);
    }

    // 6 Props
    for (let i = 0; i < 6; i++) {
      const p = createDeclaration(
        `prop${i}`,
        DeclarationType.PROP,
        '/p.ts',
        i + 1,
        1,
      );
      declarations.set(`prop${i}`, p);
      deadCode.push(p);
    }

    const result: AnalysisResult = {
      declarations,
      deadCode,
      duration: 2345,
    };

    const report = normalizeReport(buildDetailedReport(result));
    expect(report).toMatchSnapshot();
  });

  it('should generate report with deprecated declarations only', () => {
    const declarations = new Map<string, Declaration>();
    const fn = createDeclaration(
      'oldFunc',
      DeclarationType.FUNCTION,
      '/f.ts',
      1,
      1,
      false,
      false,
    );
    const cls = createDeclaration(
      'OldClass',
      DeclarationType.CLASS,
      '/c.ts',
      1,
      1,
      true,
      true,
    );
    declarations.set('fn', fn);
    declarations.set('cls', cls);

    const result: AnalysisResult = {
      declarations,
      deadCode: [fn],
      deprecated: [fn, cls],
      duration: 1000,
    };

    const report = normalizeReport(buildDetailedReport(result));
    expect(report).toMatchSnapshot();
  });

  it('should generate report with locally used export', () => {
    const declarations = new Map<string, Declaration>();
    const exp1 = createDeclaration(
      'myExport',
      DeclarationType.EXPORT,
      '/e.ts',
      10,
      5,
      true,
      false,
    );
    const exp2 = createDeclaration(
      'publicExport',
      DeclarationType.EXPORT,
      '/e.ts',
      20,
      5,
      true,
      true,
    );
    declarations.set('exp1', exp1);
    declarations.set('exp2', exp2);

    const result: AnalysisResult = {
      declarations,
      deadCode: [],
      duration: 1000,
    };

    const report = normalizeReport(buildDetailedReport(result));
    expect(report).toMatchSnapshot();
  });

  it('should generate report with deprecated and local exports', () => {
    const declarations = new Map<string, Declaration>();

    // Deprecated utilisée
    const depFunc = createDeclaration(
      'oldAPI',
      DeclarationType.FUNCTION,
      '/api.ts',
      5,
      1,
      true,
      true,
    );

    // Deprecated non utilisée
    const depClass = createDeclaration(
      'LegacyService',
      DeclarationType.CLASS,
      '/srv.ts',
      10,
      1,
      false,
      false,
    );

    // Export utilisé localement
    const localExp = createDeclaration(
      'internalHelper',
      DeclarationType.EXPORT,
      '/helper.ts',
      3,
      1,
      true,
      false,
    );

    // Export public
    const publicExp = createDeclaration(
      'publicAPI',
      DeclarationType.EXPORT,
      '/api.ts',
      20,
      1,
      true,
      true,
    );

    declarations.set('depFunc', depFunc);
    declarations.set('depClass', depClass);
    declarations.set('localExp', localExp);
    declarations.set('publicExp', publicExp);

    const result: AnalysisResult = {
      declarations,
      deadCode: [depClass],
      deprecated: [depFunc, depClass],
      duration: 1500,
    };

    const report = normalizeReport(buildDetailedReport(result));
    expect(report).toMatchSnapshot();
  });

  it('should write report to file successfully', async () => {
    const declarations = new Map<string, Declaration>();
    const fn = createDeclaration(
      'test',
      DeclarationType.FUNCTION,
      '/t.ts',
      1,
      1,
    );
    declarations.set('fn', fn);

    const result: AnalysisResult = {
      declarations,
      deadCode: [fn],
      duration: 500,
    };

    const tempDir = path.join(process.cwd(), 'temp-test-report');
    const tempFile = path.join(tempDir, 'nested', 'report.md');

    try {
      await generateFileReport(tempFile, result);

      // Vérifier que le fichier existe
      const exists = await fs
        .access(tempFile)
        .then(() => true)
        .catch(() => false);
      expect(exists).toBe(true);

      // Vérifier le contenu
      const content = await fs.readFile(tempFile, 'utf-8');
      expect(content).toContain("RAPPORT D'ANALYSE REACT-METRICS");
      expect(content).toContain('test');
    } finally {
      // Cleanup
      await fs.rm(tempDir, { recursive: true, force: true });
    }
  });

  it('should throw error when file write fails', async () => {
    const result: AnalysisResult = {
      declarations: new Map(),
      deadCode: [],
      duration: 500,
    };

    // Path invalide qui devrait échouer (ex: caractères interdits sous Windows)
    const invalidPath =
      process.platform === 'win32'
        ? 'C:\\invalid<>path\\report.md'
        : '/proc/invalid/report.md';

    await expect(generateFileReport(invalidPath, result)).rejects.toThrow();
  });
});
