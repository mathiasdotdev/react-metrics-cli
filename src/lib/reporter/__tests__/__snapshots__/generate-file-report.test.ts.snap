// Bun Snapshot v1, https://bun.sh/docs/test/snapshots

exports[`generateFileReport should generate empty report 1`] = `
"=== RAPPORT D'ANALYSE REACT-METRICS ===
GÃ©nÃ©rÃ© le: DD/MM/YYYY HH:MM:SS

=== STATISTIQUES ===
Total dÃ©clarations trouvÃ©es: 0
Code mort dÃ©tectÃ©: 0

RÃ©partition par type:

=== TOUTES LES DÃ‰CLARATIONS ===

=== CODE MORT DÃ‰TECTÃ‰ ===
Aucun code mort dÃ©tectÃ© ! ðŸŽ‰

=== DÃ‰CLARATIONS @DEPRECATED ===
Aucune dÃ©claration @deprecated dÃ©tectÃ©e."
`;

exports[`generateFileReport should generate report with functions only 1`] = `
"=== RAPPORT D'ANALYSE REACT-METRICS ===
GÃ©nÃ©rÃ© le: DD/MM/YYYY HH:MM:SS

=== STATISTIQUES ===
Total dÃ©clarations trouvÃ©es: 2
Code mort dÃ©tectÃ©: 2
Pourcentage de code mort: 100.00%

RÃ©partition par type:
  FUNCTION: 2/2 (100.0% mort)

=== TOUTES LES DÃ‰CLARATIONS ===

--- FUNCTION ---
anotherUnused | /src/other.ts:20:3 | NON UTILISÃ‰E
unusedFunction | /src/file.ts:10:5 | NON UTILISÃ‰E

=== CODE MORT DÃ‰TECTÃ‰ ===

--- FUNCTION ---
Fichier: /src/other.ts:20:3
       Nom: anotherUnused

Fichier: /src/file.ts:10:5
       Nom: unusedFunction


=== DÃ‰CLARATIONS @DEPRECATED ===
Aucune dÃ©claration @deprecated dÃ©tectÃ©e."
`;

exports[`generateFileReport should generate report with classes only 1`] = `
"=== RAPPORT D'ANALYSE REACT-METRICS ===
GÃ©nÃ©rÃ© le: DD/MM/YYYY HH:MM:SS

=== STATISTIQUES ===
Total dÃ©clarations trouvÃ©es: 2
Code mort dÃ©tectÃ©: 2
Pourcentage de code mort: 100.00%

RÃ©partition par type:
  CLASS: 2/2 (100.0% mort)

=== TOUTES LES DÃ‰CLARATIONS ===

--- CLASS ---
OldClass | /src/legacy.ts:30:7 | NON UTILISÃ‰E
UnusedClass | /src/class.ts:15:7 | NON UTILISÃ‰E

=== CODE MORT DÃ‰TECTÃ‰ ===

--- CLASS ---
Fichier: /src/legacy.ts:30:7
       Nom: OldClass

Fichier: /src/class.ts:15:7
       Nom: UnusedClass


=== DÃ‰CLARATIONS @DEPRECATED ===
Aucune dÃ©claration @deprecated dÃ©tectÃ©e."
`;

exports[`generateFileReport should generate report with constants and props 1`] = `
"=== RAPPORT D'ANALYSE REACT-METRICS ===
GÃ©nÃ©rÃ© le: DD/MM/YYYY HH:MM:SS

=== STATISTIQUES ===
Total dÃ©clarations trouvÃ©es: 2
Code mort dÃ©tectÃ©: 2
Pourcentage de code mort: 100.00%

RÃ©partition par type:
  CONSTANT: 1/1 (100.0% mort)
  PROP: 1/1 (100.0% mort)

=== TOUTES LES DÃ‰CLARATIONS ===

--- CONSTANT ---
UNUSED_CONST | /src/constants.ts:5:7 | NON UTILISÃ‰E

--- PROP ---
unusedProp | /src/props.tsx:8:3 | NON UTILISÃ‰E

=== CODE MORT DÃ‰TECTÃ‰ ===

--- CONSTANT ---
Fichier: /src/constants.ts:5:7
       Nom: UNUSED_CONST


--- PROP ---
Fichier: /src/props.tsx:8:3
       Nom: unusedProp


=== DÃ‰CLARATIONS @DEPRECATED ===
Aucune dÃ©claration @deprecated dÃ©tectÃ©e."
`;

exports[`generateFileReport should generate report with mixed categories 1`] = `
"=== RAPPORT D'ANALYSE REACT-METRICS ===
GÃ©nÃ©rÃ© le: DD/MM/YYYY HH:MM:SS

=== STATISTIQUES ===
Total dÃ©clarations trouvÃ©es: 8
Code mort dÃ©tectÃ©: 8
Pourcentage de code mort: 100.00%

RÃ©partition par type:
  CONSTANT: 1/1 (100.0% mort)
  FUNCTION: 1/1 (100.0% mort)
  CLASS: 1/1 (100.0% mort)
  CONSOLE: 1/1 (100.0% mort)
  PROP: 1/1 (100.0% mort)
  DEFINITION: 1/1 (100.0% mort)
  EXPORT: 1/1 (100.0% mort)
  DEPENDENCY: 1/1 (100.0% mort)

=== TOUTES LES DÃ‰CLARATIONS ===

--- CLASS ---
cls | /c.ts:1:1 | NON UTILISÃ‰E

--- CONSOLE ---
console | /c.ts:1:1 | NON UTILISÃ‰E

--- CONSTANT ---
const1 | /c.ts:1:1 | NON UTILISÃ‰E

--- DEFINITION ---
Type1 | /t.ts:1:1 | NON UTILISÃ‰E

--- DEPENDENCY ---
dep | /p.json:1:1 | NON UTILISÃ‰E

--- EXPORT ---
Export1 | /e.ts:1:1 | NON UTILISÃ‰E

--- FUNCTION ---
fn | /f.ts:1:1 | NON UTILISÃ‰E

--- PROP ---
prop | /p.ts:1:1 | NON UTILISÃ‰E

=== CODE MORT DÃ‰TECTÃ‰ ===

--- FUNCTION ---
Fichier: /f.ts:1:1
       Nom: fn


--- CLASS ---
Fichier: /c.ts:1:1
       Nom: cls


--- CONSTANT ---
Fichier: /c.ts:1:1
       Nom: const1


--- PROP ---
Fichier: /p.ts:1:1
       Nom: prop


--- CONSOLE ---
Fichier: /c.ts:1:1
       Nom: console


--- DEFINITION ---
Fichier: /t.ts:1:1
       Nom: Type1


--- EXPORT ---
Fichier: /e.ts:1:1
       Nom: Export1


--- DEPENDENCY ---
Fichier: /p.json:1:1
       Nom: dep


=== DÃ‰CLARATIONS @DEPRECATED ===
Aucune dÃ©claration @deprecated dÃ©tectÃ©e."
`;

exports[`generateFileReport should generate report with real-world scenario 1`] = `
"=== RAPPORT D'ANALYSE REACT-METRICS ===
GÃ©nÃ©rÃ© le: DD/MM/YYYY HH:MM:SS

=== STATISTIQUES ===
Total dÃ©clarations trouvÃ©es: 31
Code mort dÃ©tectÃ©: 31
Pourcentage de code mort: 100.00%

RÃ©partition par type:
  CONSTANT: 12/12 (100.0% mort)
  FUNCTION: 10/10 (100.0% mort)
  CLASS: 3/3 (100.0% mort)
  PROP: 6/6 (100.0% mort)

=== TOUTES LES DÃ‰CLARATIONS ===

--- CLASS ---
Class0 | /c.ts:1:1 | NON UTILISÃ‰E
Class1 | /c.ts:2:1 | NON UTILISÃ‰E
Class2 | /c.ts:3:1 | NON UTILISÃ‰E

--- CONSTANT ---
const0 | /c.ts:1:1 | NON UTILISÃ‰E
const1 | /c.ts:2:1 | NON UTILISÃ‰E
const10 | /c.ts:11:1 | NON UTILISÃ‰E
const11 | /c.ts:12:1 | NON UTILISÃ‰E
const2 | /c.ts:3:1 | NON UTILISÃ‰E
const3 | /c.ts:4:1 | NON UTILISÃ‰E
const4 | /c.ts:5:1 | NON UTILISÃ‰E
const5 | /c.ts:6:1 | NON UTILISÃ‰E
const6 | /c.ts:7:1 | NON UTILISÃ‰E
const7 | /c.ts:8:1 | NON UTILISÃ‰E
const8 | /c.ts:9:1 | NON UTILISÃ‰E
const9 | /c.ts:10:1 | NON UTILISÃ‰E

--- FUNCTION ---
function0 | /f.ts:1:1 | NON UTILISÃ‰E
function1 | /f.ts:2:1 | NON UTILISÃ‰E
function2 | /f.ts:3:1 | NON UTILISÃ‰E
function3 | /f.ts:4:1 | NON UTILISÃ‰E
function4 | /f.ts:5:1 | NON UTILISÃ‰E
function5 | /f.ts:6:1 | NON UTILISÃ‰E
function6 | /f.ts:7:1 | NON UTILISÃ‰E
function7 | /f.ts:8:1 | NON UTILISÃ‰E
function8 | /f.ts:9:1 | NON UTILISÃ‰E
function9 | /f.ts:10:1 | NON UTILISÃ‰E

--- PROP ---
prop0 | /p.ts:1:1 | NON UTILISÃ‰E
prop1 | /p.ts:2:1 | NON UTILISÃ‰E
prop2 | /p.ts:3:1 | NON UTILISÃ‰E
prop3 | /p.ts:4:1 | NON UTILISÃ‰E
prop4 | /p.ts:5:1 | NON UTILISÃ‰E
prop5 | /p.ts:6:1 | NON UTILISÃ‰E

=== CODE MORT DÃ‰TECTÃ‰ ===

--- FUNCTION ---
Fichier: /f.ts:1:1
       Nom: function0

Fichier: /f.ts:2:1
       Nom: function1

Fichier: /f.ts:3:1
       Nom: function2

Fichier: /f.ts:4:1
       Nom: function3

Fichier: /f.ts:5:1
       Nom: function4

Fichier: /f.ts:6:1
       Nom: function5

Fichier: /f.ts:7:1
       Nom: function6

Fichier: /f.ts:8:1
       Nom: function7

Fichier: /f.ts:9:1
       Nom: function8

Fichier: /f.ts:10:1
       Nom: function9


--- CLASS ---
Fichier: /c.ts:1:1
       Nom: Class0

Fichier: /c.ts:2:1
       Nom: Class1

Fichier: /c.ts:3:1
       Nom: Class2


--- CONSTANT ---
Fichier: /c.ts:1:1
       Nom: const0

Fichier: /c.ts:2:1
       Nom: const1

Fichier: /c.ts:11:1
       Nom: const10

Fichier: /c.ts:12:1
       Nom: const11

Fichier: /c.ts:3:1
       Nom: const2

Fichier: /c.ts:4:1
       Nom: const3

Fichier: /c.ts:5:1
       Nom: const4

Fichier: /c.ts:6:1
       Nom: const5

Fichier: /c.ts:7:1
       Nom: const6

Fichier: /c.ts:8:1
       Nom: const7

Fichier: /c.ts:9:1
       Nom: const8

Fichier: /c.ts:10:1
       Nom: const9


--- PROP ---
Fichier: /p.ts:1:1
       Nom: prop0

Fichier: /p.ts:2:1
       Nom: prop1

Fichier: /p.ts:3:1
       Nom: prop2

Fichier: /p.ts:4:1
       Nom: prop3

Fichier: /p.ts:5:1
       Nom: prop4

Fichier: /p.ts:6:1
       Nom: prop5


=== DÃ‰CLARATIONS @DEPRECATED ===
Aucune dÃ©claration @deprecated dÃ©tectÃ©e."
`;

exports[`generateFileReport should generate report with deprecated declarations only 1`] = `
"=== RAPPORT D'ANALYSE REACT-METRICS ===
GÃ©nÃ©rÃ© le: DD/MM/YYYY HH:MM:SS

=== STATISTIQUES ===
Total dÃ©clarations trouvÃ©es: 2
Code mort dÃ©tectÃ©: 1
Pourcentage de code mort: 50.00%

RÃ©partition par type:
  FUNCTION: 1/1 (100.0% mort)
  CLASS: 0/1 (0.0% mort)

DÃ©clarations @deprecated: 2

=== TOUTES LES DÃ‰CLARATIONS ===

--- CLASS ---
OldClass | /c.ts:1:1 | UTILISÃ‰E

--- FUNCTION ---
oldFunc | /f.ts:1:1 | NON UTILISÃ‰E

=== CODE MORT DÃ‰TECTÃ‰ ===

--- FUNCTION ---
Fichier: /f.ts:1:1
       Nom: oldFunc


=== DÃ‰CLARATIONS @DEPRECATED ===
Suggestion: ConsidÃ©rer la suppression ou le remplacement de ces dÃ©clarations.


--- FUNCTION ---
Fichier: /f.ts:1:1
       Nom: oldFunc
       Statut: NON UTILISÃ‰E


--- CLASS ---
Fichier: /c.ts:1:1
       Nom: OldClass
       Statut: UTILISÃ‰E
"
`;

exports[`generateFileReport should generate report with locally used export 1`] = `
"=== RAPPORT D'ANALYSE REACT-METRICS ===
GÃ©nÃ©rÃ© le: DD/MM/YYYY HH:MM:SS

=== STATISTIQUES ===
Total dÃ©clarations trouvÃ©es: 2
Code mort dÃ©tectÃ©: 0
Pourcentage de code mort: 0.00%

RÃ©partition par type:
  EXPORT: 0/2 (0.0% mort)

=== TOUTES LES DÃ‰CLARATIONS ===

--- EXPORT ---
myExport | /e.ts:10:5 | UTILISÃ‰E LOCALEMENT (supprimer export)
publicExport | /e.ts:20:5 | UTILISÃ‰E

=== CODE MORT DÃ‰TECTÃ‰ ===
Aucun code mort dÃ©tectÃ© ! ðŸŽ‰

=== DÃ‰CLARATIONS @DEPRECATED ===
Aucune dÃ©claration @deprecated dÃ©tectÃ©e."
`;

exports[`generateFileReport should generate report with deprecated and local exports 1`] = `
"=== RAPPORT D'ANALYSE REACT-METRICS ===
GÃ©nÃ©rÃ© le: DD/MM/YYYY HH:MM:SS

=== STATISTIQUES ===
Total dÃ©clarations trouvÃ©es: 4
Code mort dÃ©tectÃ©: 1
Pourcentage de code mort: 25.00%

RÃ©partition par type:
  FUNCTION: 0/1 (0.0% mort)
  CLASS: 1/1 (100.0% mort)
  EXPORT: 0/2 (0.0% mort)

DÃ©clarations @deprecated: 2

=== TOUTES LES DÃ‰CLARATIONS ===

--- CLASS ---
LegacyService | /srv.ts:10:1 | NON UTILISÃ‰E

--- EXPORT ---
internalHelper | /helper.ts:3:1 | UTILISÃ‰E LOCALEMENT (supprimer export)
publicAPI | /api.ts:20:1 | UTILISÃ‰E

--- FUNCTION ---
oldAPI | /api.ts:5:1 | UTILISÃ‰E

=== CODE MORT DÃ‰TECTÃ‰ ===

--- CLASS ---
Fichier: /srv.ts:10:1
       Nom: LegacyService


=== DÃ‰CLARATIONS @DEPRECATED ===
Suggestion: ConsidÃ©rer la suppression ou le remplacement de ces dÃ©clarations.


--- FUNCTION ---
Fichier: /api.ts:5:1
       Nom: oldAPI
       Statut: UTILISÃ‰E


--- CLASS ---
Fichier: /srv.ts:10:1
       Nom: LegacyService
       Statut: NON UTILISÃ‰E
"
`;
